packages/stark-ui/src/modules/action-bar/components/action-bar.component.spec.ts

```diff
@@ -0,0 +1,140 @@
+import { STARK_LOGGING_SERVICE, STARK_ROUTING_SERVICE } from "@nationalbankbelgium/stark-core";
+import { MockStarkLoggingService, MockStarkRoutingService } from "@nationalbankbelgium/stark-core/testing";
+import { StarkSvgViewBoxModule } from "../../svg-view-box/svg-view-box.module";
+import { StarkActionBarComponent } from "./action-bar.component";
+import {StarkActionBarConfig} from "./action-bar-config.intf";
+import {StarkAction} from "./action.intf";
+
+describe("ActionBarComponent", () => {
+	let fixture: ComponentFixture<StarkActionBarComponent>;
+	let component: StarkActionBarComponent;
+	const buttonToggleSelector: string = ".extend-action-bar";
+	
+	beforeEach(async(() => {
+		return (
+			TestBed.configureTestingModule({
+				declarations: [StarkActionBarComponent],
+				imports: [
+					StarkSvgViewBoxModule,
+					MatButtonModule,
+					MatMenuModule,
+					MatTooltipModule
+				],
+				providers: [
+					{ provide: STARK_LOGGING_SERVICE, useValue: new MockStarkLoggingService() },
+					{ provide: STARK_ROUTING_SERVICE, useClass: MockStarkRoutingService }
+				],
+				schemas: [NO_ERRORS_SCHEMA] // tells the Angular compiler to ignore unrecognized elements and attributes
+			}).compileComponents()
+		);
+	}));
+
+	
+	beforeEach(() => {
+		fixture = TestBed.createComponent(StarkActionBarComponent);
+		component = fixture.componentInstance;
+		const demoActions: Array<StarkAction> = [
+			{
+				id: "userDetailValidate",
+				label: "Validate",
+				icon: "check",
+				actionCall: ($event: Event, data: any): void => {
+					console.log($event);
+					console.log(data);
+				},
+				isEnabled: false,
+				isVisible: true
+			},
+			{
+				id: "userDetailSave",
+				label: "Save",
+				icon: "content-save",
+				actionCall: ($event: Event, data: any): void => {
+					console.log($event);
+					console.log(data);
+				},
+				isEnabled: true,
+				isVisible: true
+			}
+		];
+
+		const demoActionBarConfig: StarkActionBarConfig = {
+			actions: demoActions,
+			isPresent: true
+		};
+
+		component.actionBarConfig = demoActionBarConfig;
+		fixture.detectChanges();
+	});
+
+	describe("@Input() mode", () => {
+		it("should have the toggle action bar button visible in full mode", () => {
+			component.mode = 'full';
+			fixture.detectChanges();
+			const buttonToggleExtend: HTMLElement = fixture.nativeElement.querySelector(buttonToggleSelector);
+			expect(buttonToggleExtend).toBeTruthy();
+		});
+
+		it("should not have the toggle action bar button visible in compact mode", () => {
+			component.mode = 'compact';
+			fixture.detectChanges();
+			const buttonToggleExtend: HTMLElement = fixture.nativeElement.querySelector(buttonToggleSelector);
+			expect(buttonToggleExtend).toBeFalsy();
+		});
+	});
+
+	describe("@Input() actionBarId", () => {
+		it("should have set the id of the action bar", () => {
+			const actionBar: HTMLElement = fixture.nativeElement.querySelector("#"+component.actionBarId);
+			component.actionBarId = "action-bar-id";
```

Comment:
These 2 lines are a bit weird and confusing. The first line sets already the id (which is unknown by the way at this point) and then the second line sets the id. Finally in the last line of this test the id is checked.

You can improve this test by one of these 2 options:
- changing the order of these 2 lines and get the HTML element just after calling `fixture.detectChanges()`
- add an expectation just after the first line to expect the HTMLElement to be undefined (because the querySelector will not find the element) and then add another expectation before the last line to expect the HTMLElement to be defined.